bambi/plots/plot_cap.py

```diff
@@ -0,0 +1,253 @@
+    else:
+        data_dict = {main: main_values}
+
+    # Construct dictionary of terms that are in the model
+    terms = {}
+    if model._design.common:
+        terms.update(model._design.common.terms)
+
+    if model._design.group:
+        terms.update(model._design.group.terms)
+
+    # Get default values for each variable in the model
+    for term in terms.values():
+        if hasattr(term, "components"):
+            for component in term.components:
+                name = component.name
+                if name not in data_dict:
+                    # For numeric predictors, select the mean.
+                    if component.kind == "numeric":
+                        data_dict[name] = np.mean(data[name])
+                    # For categoric predictors, select the most frequent level.
+                    elif component.kind == "categoric":
+                        data_dict[name] = mode(data[name])
+
+    cap_data = pd.DataFrame(data_dict)
+
+    # Make sure new types are same types than the original columns
+    for column in cap_data:
+        cap_data[column] = cap_data[column].astype(data[column].dtype)
+    return cap_data
+
+
+def plot_cap(model, idata, covariates, hdi_prob=0.94, legend=True, ax=None):
+    """Plot Conditional Adjusted Predictions
+
+    Parameters
+    ----------
+    model : bambi.Model
+        The model for which we want to plot the predictions.
+    idata : arviz.InferenceData
+        The InferenceData object that contains the samples from the posterior distribution of
+        the model.
+    covariates : list
+        A sequence of one or two names of variables. The first variable is taken as the main
+        variable. If present, the second variable is a grouping variable.
+    hdi_prob : float, optional
+        The probability for the credibility intervals. Must be between 0 and 1. Defaults to 0.94.
+    legend : bool, optional
+        Whether to automatically include a legend in the plot. Defaults to ``True``.
+    ax : matplotlib.axes._subplots.AxesSubplot, optional
+        A matplotlib axes object. If None, this function instantiates a new axes object.
+        Defaults to ``None``.
+
+    Returns
+    -------
+    matplotlib.figure.Figure, matplotlib.axes._subplots.AxesSubplot
+        A tuple with the figure and the axes.
+
+    Raises
+    ------
+    ValueError
+        When ``level`` is not within 0 and 1.
+        When the main covariate is not numeric or categoric.
+    """
+
+    covariates = listify(covariates)
+    assert len(covariates) in [1, 2]
+
+    cap_data = create_cap_data(model, covariates)
+    idata = model.predict(idata, data=cap_data, inplace=False)
+
+    if not 0 < hdi_prob < 1:
+        raise ValueError(f"'level' must be greater than 0 and smaller than 1. It is {hdi_prob}.")
+
+    lower_bound = round((1 - hdi_prob) / 2, 4)
+    upper_bound = 1 - lower_bound
+
+    y_hat = idata.posterior[f"{model.response.name}_mean"]
```

Comment:
Love plot_cap so far! One potential issue here: after using plot_cap once, I realized the function adds 200 (_grid_n_) new entries into the idata's posterior data variables. The issue is, when running az.summary(idata) after plot_cap, it will include all of those y_hat_mean values (as would any other az plot like plot_trace).

Reply:
Oh! That should be a problem with the `.predict()` method ignoring the `inplace=False` argument. Could you please open an issue with a minimum reproducible example that shows the problem? Thanks @vishalthatsme !